doc/esmvalcore/preprocessor.rst

```diff
@@ -3,41 +3,75 @@
 ************
 Preprocessor
 ************
-The ESMValTool preprocessor can be used to perform a broad range of operations
-on the input data before diagnostics or metrics are applied. The
-preprocessor performs these operations in a centralized, documented and
-efficient way, thus reducing the data processing load on the diagnostics side.
-
-Each of the preprocessor operations is written in a dedicated python module and
-all of them receive and return an Iris cube, working sequentially on the data
-with no interactions between them.  The order
-in which the preprocessor operations is applied is set by default in order to
-minimize the loss of information due to, for example, temporal and spatial
-subsetting or multi-model averaging. Nevertheless, the user is free to change
-such order to address specific scientific requirements, but keeping in mind
-that some operations must be necessarily performed in a specific order. This is
-the case, for instance, for multi-model statistics, which required the model to
-be on a common grid and therefore has to be called after the regridding module.
 
 In this section, each of the preprocessor modules is described in detail
 following the default order in which they are applied:
 
-* `Variable derivation`_.
-* `CMOR check and dataset-specific fixes`_.
-* `Vertical interpolation`_.
-* `Land/Sea/Ice masking`_.
-* `Horizontal regridding`_.
-* `Masking of missing values`_.
-* `Multi-model statistics`_.
-* `Time operations`_.
-* `Area operations`_.
-* `Volume operations`_.
-* `Unit conversion`_.
+* :ref:`Variable derivation`
+* :ref:`CMOR check and dataset-specific fixes`
+* :ref:`Vertical interpolation`
+* :ref:`Land/Sea/Ice masking`
+* :ref:`Horizontal regridding`
+* :ref:`Masking of missing values`
+* :ref:`Multi-model statistics`
+* :ref:`Time operations`
+* :ref:`Area operations`
+* :ref:`Volume operations`
+* :ref:`Unit conversion`
+
+Overview
+========
+
+.. 
+   ESMValTool is a modular ``Python 3.6+`` software package possesing capabilities
+   of executing a large number of diagnostic routines that can be written in a
+   number of programming languages (Python, NCL, R, Julia). The modular nature
+   benefits the users and developers in different key areas: a new feature
+   developed specifically for version 2.0 is the preprocessing core  or the
+   preprocessor (esmvalcore) that executes the bulk of standardized data
+   operations and is highly optimized for maximum performance in data-intensive
+   tasks. The main objective of the preprocessor is to integrate as many
+   standardizable data analysis functions as possible so that the diagnostics can
+   focus on the specific scientific tasks they carry. The preprocessor is linked
+   to the diagnostics library and the diagnostic execution is seamlessly performed
+   after the preprocessor has completed the its steps. The benefit of having a
```

Comment:
Remove the "the" after "completed"?

Reply:
This part is commented out (using `..` at line 25) since I think it does not fit to this section.
Still, I did not want to remove the text as it may be useful for other parts of the docu.

Reply:
Anyway I fixed the typo...